internal/cgroup/cgroup.go

```diff
@@ -0,0 +1,52 @@
+package cgroup
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	cgroupRoot = "/sys/fs/cgroup"
+	basePath   = "jobsvc"
+)
+
+func BasePath() string {
+	return filepath.Join(cgroupRoot, basePath)
+}
+
+func Create(jobId string) error {
+	groupPath := filepath.Join(BasePath(), jobId)
+
+	if err := os.MkdirAll(groupPath, 0755); err != nil {
+		return fmt.Errorf("failed to create cgroup: %w", err)
+	}
+
+	limits := map[string]string{
+		"cpu.max":    "50000 100000",
+		"memory.max": "1073741824",
+		"io.max":     "default rbps=104857600 wbps=104857600",
+	}
+
+	for file, value := range limits {
+		path := filepath.Join(groupPath, file)
+		if err := os.WriteFile(path, []byte(value), 0644); err != nil {
+			return fmt.Errorf("failed to set %s: %w", file, err)
+		}
+	}
+
+	return nil
+}
+
+func Remove(jobId string) error {
+	groupPath := filepath.Join(BasePath(), jobId)
+	return os.RemoveAll(groupPath)
```

Comment:
This only works because `RemoveAll` tries a `Remove` first, it'll fail with a lot of confusing errors as it tries to delete all the special files in the cgroup directory if the cgroup can't be removed, might as well do just the one remove.

```suggestion
	return os.Remove(groupPath)
```